moabb/evaluations/splitters.py

```diff
@@ -267,3 +267,124 @@ def split(self, y, metadata):
                 yield subject_indices[train_session_idx], subject_indices[
                     test_session_idx
                 ]
+
+
+class CrossSubjectSplitter(BaseCrossValidator):
+    """Data splitter for cross subject evaluation.
+
+    This splitter enables cross-subject evaluation by performing a Leave-One-Session-Out (LOSO)
+    cross-validation on the dataset.
+
+    It assumes that the entire metainformation across all subjects is already loaded.
+
+    Unlike the `CrossSubjectEvaluation` class from `moabb.evaluation`, which manages
+    the complete evaluation process end-to-end, this splitter is solely responsible
+    for dividing the data into training and testing sets based on subjects.
+
+    .. image:: https://raw.githubusercontent.com/NeuroTechX/moabb/refs/heads/develop/docs/source/images/crosssubj.png
+        :alt: The schematic diagram of the CrossSubject split
+        :align: center
+
+    The inner cross-validation strategy can be changed by passing the
+    `cv_class` and `cv_kwargs` arguments. By default, it uses LeaveOneGroupOut,
+    which performs Leave-One-Subject-Out cross-validation.
+
+    Parameters
+    ----------
+    cv_class: cross-validation class, default=LeaveOneGroupOut
+        Inner cross-validation strategy for splitting the sessions of one subject.
+        LeaveOneGroupOut is the most common default.
+    shuffle: bool, default=False
+        Whether to shuffle the subject order. It can only be
+        used when changing the `cv_class` to a class compatible with `shuffle`.
+    random_state: int, RandomState instance or None, default=None
+        Controls the randomness of the inner cross-validation when `shuffle` is True.
+        Pass an int for reproducible output across multiple function calls.
+        For `cv_class` accepting `random_state`, they are provided with a shared rng.
+    cv_kwargs: dict
+        Additional arguments to pass to the inner cross-validation strategy.
+
+    Yields
+    ------
+    train : ndarray
+        The training set indices for that split.
+
+    test : ndarray
+        The testing set indices for that split.
+    """
+
+    def __init__(
+        self,
+        cv_class: type[BaseCrossValidator] = LeaveOneGroupOut,
+        shuffle: bool = False,
+        random_state: int = None,
+        **cv_kwargs,
+    ):
+        self.cv_class = cv_class
+        self.cv_kwargs = cv_kwargs
+        self._cv_kwargs = dict(**cv_kwargs)
+
+        self.shuffle = shuffle
+        self.random_state = random_state
+        self._rng = check_random_state(random_state) if shuffle else None
```

Comment:
Here, we only have one CV object, so we should pass the value of `random_state` directly to it.
If we create a `_rng`, this would yield different splits each time we call the same object again, or even if we call `get_n_splits`.

Reply:
good point @tomMoral